sw/device/silicon_creator/lib/log.c

```diff
@@ -0,0 +1,45 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/log.h"
+
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/drivers/uart.h"
+
+rom_error_t log_printf(const char *format, ...) {
+  va_list args;
+  va_start(args, format);
+
+  while (*format != '\0') {
+    if (*format == '%') {
+      ++format;
+      switch (*format++) {
+        case 's': {
+          // print string
+          const char *str = va_arg(args, const char *);
+          while (*str != '\0') {
+            uart_putchar(*str++);
+          }
+          continue;
+        }
+        case 'x': {
+          // print hex value
+          const char kHexTable[16] = "0123456789abcdef";
+          uint32_t v = va_arg(args, uint32_t);
+          for (int i = 28; i >= 0; i -= 4) {
```

Comment:
nit: s/int/size_t/ for consistency. Even though we are not super consistent :). 

Unrolling this operation is also fine. It may make things easier to read.

Reply:
sg, thanks!